vyper/semantics/validation/annotation.py

```diff
@@ -173,9 +181,23 @@ def visit_Constant(self, node, type_):
     def visit_Dict(self, node, type_):
         node._metadata["type"] = type_
 
+    def visit_Hex(self, node, type_):
+        node._metadata["type"] = type_
+
     def visit_Index(self, node, type_):
         self.visit(node.value, type_)
 
+    def visit_Int(self, node, type_):
+        if type_:
+            if not hasattr(type_, "_id"):
+                # Filter for IntegerAbstractType and set it to the largest unsigned type
```

Comment:
seems out of line with the other implementations here. is this dead code?

Reply:
This is still required for arrays because the index type for `_SequenceDefinition` is `IntegerAbstractType`.

Reply:
Overriding `get_index_type` for `_SequenceDefinition` to be `uint256`, similar to `get_signature`.